src/HealthChecks.Redis/RedisHealthCheck.cs

```diff
@@ -81,4 +80,21 @@ public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context
             return new HealthCheckResult(context.Registration.FailureStatus, exception: ex);
         }
     }
+
+    private static async Task<ConnectionMultiplexer> TimeoutAsync(Task<ConnectionMultiplexer> task, CancellationToken cancellationToken)
+    {
+        using var timeoutCts = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
+        var completedTask = await Task.
+            WhenAny(task, Task.Delay(Timeout.Infinite, timeoutCts.Token))
+            .ConfigureAwait(false);
+
+        if (completedTask == task)
+        {
+            timeoutCts.Cancel();
+            return await task.ConfigureAwait(false);
+        }
+
+        cancellationToken.ThrowIfCancellationRequested();
+        throw new OperationCanceledException();

```

Comment:
I still needed a return value here, so I decided to throw the same exception that `ThrowIfCancellationRequested()` would throw.
Technically we should never hit this line because if we're at line 97, that cancellation token has already timed out.